R/metab_simple.R

```diff
@@ -48,31 +54,39 @@ metab_simple <- function(data, ...) {
   #' 
   #' @param params a vector of length 2, where the first element is GPP and the second element is ER (both mg/L/d)
   #' @param 
-#   onestation_likelihood <- function(params, DO, DO.deficit, oxy, light, z, bp, ts, K) {
-#     
-#     GPP <- params[1]
-#     ER <- params[2]
-#     
-#     DO.modeled <- numeric(nrow(data))
-#     DO.modeled[1] <- oxy[1]
-#     for (i in 2:length(DO)) {
-#       DO.modeled[i] <- 
-#         DO.modeled[i-1] +
-#         (GPP/z)*(light[i]/sum(light)) + 
-#         ER * ts / z + 
-#         Kcor(temp[i],K)*ts*DO.deficit 
-#     }
-#   
-#     ##below is MLE calculation
-#     sqdiff<-(oxy-metab)^2 
-#     length(oxy)*(log(((sum(sqdiff)/length(oxy))^0.5)) +0.5*log(6.28))   + ((2*sum(sqdiff)/length(oxy))^-1)*sum(sqdiff)
-#   }
-  
+  onestation_negloglik <- function(params, DO.obs, DO.deficit, depth, k.O2, frac.GPP, frac.ER, frac.D) {
+    
+    # Parse params vector (passed from nlm)
+    GPP <- params[1]
+    ER <- params[2]
+    
+    # Model DO with given params
+    DO.mod <- numeric(length(DO.obs))
+    DO.mod[1] <- DO.obs[1]
+    DO.delta <- 
+      GPP * frac.GPP / depth + 
+      ER * frac.ER / depth + 
+      k.O2 * DO.deficit * frac.D # Bob - pros & cons of using DO.mod[t-1] instead of DO.obs[t] or DO.obs[t-1] (i.e., from DO.deficit) here?
```

Comment:
Alison,

Yep, I am getting the emails.  Yall know its after 6 there right?

I calc DOdef as DOsat-mod.DO[t-1] to keep the data out of the model.  I am not sure we want to confront the model with data used in the model structure itself.  I see that Van de Bogert does use the actual data  thier 2012 paper.  If we make this change it will require extensive checking.  I am not convinced we want to do it this way.

Bob
On Apr 24, 2015, at 2:56 PM, Alison Appling <notifications@github.com<mailto:notifications@github.com>> wrote:


In R/metab_simple.R<https://github.com/USGS-R/streamMetabolizer/pull/25#discussion_r29085089>:

> -#     length(oxy)*(log(((sum(sqdiff)/length(oxy))^0.5)) +0.5*log(6.28))   + ((2*sum(sqdiff)/length(oxy))^-1)*sum(sqdiff)
> -#   }
> -
> +  onestation_negloglik <- function(params, DO.obs, DO.deficit, depth, k.O2, frac.GPP, frac.ER, frac.D) {
> +
> +    # Parse params vector (passed from nlm)
> +    GPP <- params[1]
> +    ER <- params[2]
> +
> +    # Model DO with given params
> +    DO.mod <- numeric(length(DO.obs))
> +    DO.mod[1] <- DO.obs[1]
> +    DO.delta <-
> +      GPP * frac.GPP / depth +
> +      ER * frac.ER / depth +
> +      k.O2 * DO.deficit * frac.D # Bob - pros & cons of using DO.mod[t-1] instead of DO.obs[t] or DO.obs[t-1] (i.e., from DO.deficit) here?


@robohall<https://github.com/robohall> - this function very nearly copies your rivermetab() with onestationmle(), though with modified variable names and formatting. The one (I think) substantive change is in this line 69, where I opted to use a pre-calculated DO.deficit (calculated oxygen saturation concentration minus observed oxygen concentration, both at time i) instead of computing the difference between osat[i] (calculated oxygen saturation concentration) and metab[i-1] (modeled dissolved oxygen at previous timestep). In other words, the original code, (osat(temp[i],bp)-metab[i-1]), is replaced by DO.deficit above.

I'd like to hear your take on this - is there a strong case for using DO.mod(modeled) rather than DO.obs(observed), and/or for using DO at the previous timestep and DO.sat at the present timestep, to calculate the oxygen deficit?

—
Reply to this email directly or view it on GitHub<https://github.com/USGS-R/streamMetabolizer/pull/25/files#r29085089>.